OpenLaTeXStudio/Editor/src/main/java/latexstudio/editor/DbxFileActions.java

```diff
@@ -0,0 +1,115 @@
+import java.util.List;
+import javax.swing.DefaultListSelectionModel;
+import javax.swing.JList;
+import javax.swing.JOptionPane;
+import latexstudio.editor.files.FileService;
+import latexstudio.editor.remote.DbxEntryDto;
+import latexstudio.editor.remote.DbxState;
+import latexstudio.editor.remote.DbxUtil;
+import latexstudio.editor.util.ApplicationUtils;
+import org.apache.commons.io.IOUtils;
+import org.openide.util.Exceptions;
+
+public class DbxFileActions {
+
+    private EditorTopComponent etc;
+
+    public DbxFileActions(EditorTopComponent etc) {
+        this.etc = etc;
+    }
+
+    public void saveProgress(DropboxRevisionsTopComponent drtc) {
+        DbxClient client = DbxUtil.getDbxClient();
+
+        if (client == null) {
+            return;
+        }
+
+        String sourceFileName = ApplicationUtils.getTempSourceFile();
+        File file = new File(sourceFileName);
+
+        FileInputStream inputStream = null;
+        try {
+            inputStream = new FileInputStream(file);
+        } catch (FileNotFoundException ex) {
+            Exceptions.printStackTrace(ex);
+        }
+
+        DbxState dbxState = etc.getDbxState();
+
+        if (dbxState != null) {
+            try {
+                DbxEntry.File uploadedFile = client.uploadFile(dbxState.getPath(),
+                        DbxWriteMode.update(dbxState.getRevision()), file.length(), inputStream);
+                JOptionPane.showMessageDialog(null,
+                        "Successfuly updated file " + uploadedFile.name + " (" + uploadedFile.humanSize + ")",
+                        "File updated in Dropbox",
+                        JOptionPane.INFORMATION_MESSAGE);
+                drtc.updateRevisionsList(uploadedFile.path);
+                etc.setDbxState(new DbxState(uploadedFile.path, uploadedFile.rev));
+            } catch (DbxException ex) {
+                DbxUtil.showDbxAccessDeniedPrompt();
+            } catch (IOException ex) {
+                Exceptions.printStackTrace(ex);
+            } finally {
+                IOUtils.closeQuietly(inputStream);
+                etc.setModified(false);
+            }
+        } else {
+            JOptionPane.showMessageDialog(null, "No Dropbox file has been loaded.\n"
+                    + "You must open Dropbox file, before you save it.", "Cannot save progress", JOptionPane.WARNING_MESSAGE);
+        }
+    }
+
+    public void openFromDropbox(DropboxRevisionsTopComponent drtc, RevisionDisplayTopComponent revtc) {
+        DbxClient client = DbxUtil.getDbxClient();
+
+        if (client == null) {
+            return;
+        }
+
+        List<DbxEntryDto> dbxEntries = new ArrayList<>();
+
+        try {
+            for (DbxEntry entry : client.searchFileAndFolderNames("/", ".tex")) {
+                dbxEntries.add(new DbxEntryDto(entry));
+            }
+        } catch (DbxException ex) {
+            DbxUtil.showDbxAccessDeniedPrompt();
+            return; //No point continuing
```

Comment:
Please refactor this as a method
  List<DbxEntryDto> getDbxTexEntries()
 
Return an empty list, in case of DbxException
Check if the list is empty an message to user that there are no *.tex-Files found.